pybutton/request.py

```diff
@@ -117,9 +123,24 @@ def request_url(secure, hostname, port, path):
         Returns:
             (str) A complete url made up of the arguments.
     '''
+    query = urlencode(query) if query else ''
```

Comment:
Also consider calling your local variable something other than `query` to avoid hiding the argument variable.